### PR TITLE
Added unit tests for treelite model.

### PR DESCRIPTION
**Pull request by cnv1989**
This PR adds unit tests for treelite model.
Output of ./dlr_treelite_test
```bash
[==========] Running 9 tests from 1 test case.
[----------] Global test environment set-up.
[----------] 9 tests from TreeliteTest
[ RUN      ] TreeliteTest.TestGetNumInputs
[       OK ] TreeliteTest.TestGetNumInputs (29 ms)
[ RUN      ] TreeliteTest.TestGetInputName
[       OK ] TreeliteTest.TestGetInputName (18 ms)
[ RUN      ] TreeliteTest.TestGetInputType
[       OK ] TreeliteTest.TestGetInputType (18 ms)
[ RUN      ] TreeliteTest.TestGetInput
[       OK ] TreeliteTest.TestGetInput (29 ms)
[ RUN      ] TreeliteTest.TestGetNumOutputs
[       OK ] TreeliteTest.TestGetNumOutputs (13 ms)
[ RUN      ] TreeliteTest.TestGetOutputType
[       OK ] TreeliteTest.TestGetOutputType (20 ms)
[ RUN      ] TreeliteTest.TestGetOutputShape
[       OK ] TreeliteTest.TestGetOutputShape (19 ms)
[ RUN      ] TreeliteTest.TestGetOutputSizeDim
[       OK ] TreeliteTest.TestGetOutputSizeDim (29 ms)
[ RUN      ] TreeliteTest.TestGetOutput
[       OK ] TreeliteTest.TestGetOutput (15 ms)
[----------] 9 tests from TreeliteTest (194 ms total)

[----------] Global test environment tear-down
[==========] 9 tests from 1 test case ran. (194 ms total)
[  PASSED  ] 9 tests.
```

Thanks for contributing to DLR! By submitting this pull request, you confirm that your contribution is made under the terms of the [Apache 2.0 license](https://github.com/neo-ai/neo-ai-dlr/blob/master/LICENSE).

Please refer to our [guideline](https://github.com/neo-ai/neo-ai-dlr/blob/master/CONTRIBUTING.md) for useful information and tips.

### CODE REVIEW

File: src/dlr_treelite.cc [hunk 1]
```diff
@@ -171,6 +171,7 @@ void TreeliteModel::GetOutputShape(int index, int64_t* shape) const {
 
 void TreeliteModel::GetOutput(int index, void* out) {
   CHECK(treelite_input_);
+  std::cout << "Number of output data: " << treelite_output_.data() << std::endl;
```
Comment (cnv1989) [inline src/dlr_treelite.cc:174]: remove this.

File: tests/cpp/dlr_test.cc [hunk 2]
```diff
@@ -73,7 +73,7 @@ TEST(DLR, TestGetDLROutputShape) {
   int index = 0;
   EXPECT_EQ(GetDLROutputSizeDim(&model, 0, &output_size, &output_dim), 0);
   EXPECT_EQ(output_dim, 1);
-  
+
```
Comment (cnv1989) [inline tests/cpp/dlr_test.cc:76]: remove this.

File: tests/cpp/dlr_treelite_test.cc [hunk 3]
```diff
@@ -11,3 +8,93 @@ int main(int argc, char** argv) {
...
+  int output_dim;
+  EXPECT_NO_THROW(model->GetOutputSizeDim(0, &output_size, &output_dim));
+  EXPECT_EQ(output_size, out_size);
+  EXPECT_EQ(output_dim, out_dim);
+}
+
+TEST_F(TreeliteTest, TestGetOutput) {
+  EXPECT_NO_THROW(model->SetInput("data", in_shape, data, in_dim));
+  EXPECT_NO_THROW(model->Run());
+  float* output = (float*) malloc(sizeof(float)*out_size);
```
Comment (apivovarov) [inline tests/cpp/dlr_treelite_test.cc:97]: out_size is just 1. Do we want to malloc 4 bytes? Simple `float output[1];` should work.

Comment (cnv1989) [reply tests/cpp/dlr_treelite_test.cc:97]: Yep makes sense.

File: tests/cpp/dlr_treelite_test.cc [hunk 4]
```diff
@@ -11,3 +8,92 @@ int main(int argc, char** argv) {
...
+    const int64_t in_shape[2] = {1, 69};
+    const int64_t out_size = 1;
+    const int out_dim = 2;
+
+    dlr::TreeliteModel* model;
+
+    TreeliteTest() {
+      // Setup input data
+      data = new float[in_size];
+      for(auto i = 0; i < in_size; i++) {
```
Comment (apivovarov) [inline tests/cpp/dlr_treelite_test.cc:26]: clang-format please

Comment (cnv1989) [reply tests/cpp/dlr_treelite_test.cc:26]: +1

**Comment by apivovarov** [comment]
Can you add `./dlr_treelite_test` output to the description?

**Comment by cnv1989** [comment]
> Can you add `./dlr_treelite_test` output to the description?

Done!

**Pull request merged**